geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/strategies/IdFilterStrategy.scala

```diff
@@ -17,20 +17,22 @@ import org.opengis.filter.{And, Filter, Id, Or}
 
 trait IdFilterStrategy[T, U] extends GeoMesaFeatureIndex[T, U] {
 
-  override def getFilterStrategy(filter: Filter,
-                                 transform: Option[SimpleFeatureType],
-                                 stats: Option[GeoMesaStats]): Option[FilterStrategy] = {
+  override def getFilterStrategy(
+      filter: Filter,
+      transform: Option[SimpleFeatureType],
+      stats: Option[GeoMesaStats]): Option[FilterStrategy] = {
     if (filter == Filter.INCLUDE) {
-      Some(FilterStrategy(this, None, None, Long.MaxValue))
+      Some(FilterStrategy(this, None, None, temporal = false, Long.MaxValue))
     } else if (filter == Filter.EXCLUDE) {
       None
     } else {
       val (ids, notIds) = IdExtractingVisitor(filter)
       if (ids.isDefined) {
         // top-priority index - always 1 if there are actually ID filters
-        Some(FilterStrategy(this, ids, notIds, IdFilterStrategy.StaticCost))
+        // note: although there's no temporal predicate, there's an implied exact date for the given feature
+        Some(FilterStrategy(this, ids, notIds, temporal = true, IdFilterStrategy.StaticCost))
```

Comment:
Nice corner case.  

This does seem to suggest that we may want to think through all the cases our query planning handles at some point and think through it holistically.  

It seems good and bad to associate an id query (which has the ids specified) as 'temporal'.  Locally, right now, it is the Right Thing though.

Reply:
yeah, the current approach of using costs and delegating the cost to the index means that we don't have a single place to rank everything... but the tradeoff is that it's extensible.